lhotse/recipes/xbmu_amdo31.py

```diff
@@ -0,0 +1,126 @@
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'recordings' and 'supervisions'.
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+    transcript_path = corpus_dir / "data/transcript/transcript_clean.txt"
+    transcript_dict = {}
+    with open(transcript_path, "r", encoding="utf-8") as f:
+        for line in f.readlines():
+            idx_transcript = line.split()
+            content = " ".join(idx_transcript[1:])
+            transcript_dict[idx_transcript[0]] = content
+    manifests = defaultdict(dict)
+    dataset_parts = ["train", "dev", "test"]
+    for part in tqdm(
+        dataset_parts,
+        desc="Process xbmu_amdo31 audio.",
+    ):
+        logging.info(f"Processing xbmu_amdo31 subset: {part}")
+        # Generate a mapping: utt_id -> (audio_path, audio_info, speaker, text)
+        recordings = []
+        supervisions = []
+        wav_path = corpus_dir / "data" / "wav" / f"{part}"
+        count = 0
+        for audio_path in wav_path.rglob("**/*.wav"):
+            idx = audio_path.stem.split("-")[1]
+            speaker = audio_path.parts[-2]
+            if idx not in transcript_dict:
+                logging.warning(f"No transcript: {idx}")
+                logging.warning(f"{audio_path} has no transcript.")
+                continue
+            text = transcript_dict[idx]
+            if not audio_path.is_file():
+                logging.warning(f"No such file: {audio_path}")
+                continue
+            recording = Recording.from_file(audio_path)
+            recordings.append(recording)
+            count += 1
+            segment = SupervisionSegment(
+                id=str(count) +  "_" + idx,
+                recording_id=speaker + "-" +idx,
+                start=0.0,
+                duration=recording.duration,
+                channel=0,
+                language="tibetan",
+                speaker=speaker,
+                text=text.strip(),
+            )
+            supervisions.append(segment)
+
+        recording_set = RecordingSet.from_recordings(recordings)
+        supervision_set = SupervisionSet.from_segments(supervisions)
+        validate_recordings_and_supervisions(recording_set, supervision_set)
+
+        if output_dir is not None:
+            supervision_set.to_file(
+                output_dir / f"xbmu_amdo31_supervisions_{part}.jsonl.gz"
+            )
+            recording_set.to_file(output_dir / f"xbmu_amdo31_recordings_{part}.jsonl.gz")
+
+        manifests[part] = {"recordings": recording_set, "supervisions": supervision_set}
+
+    return manifests
```

Comment:
Probably should have a newline at end of file to avoid flake8 error.

Reply:
Thanks for the reminder